### PR TITLE
Create a common update group supporting self-update options (#103)

### PR DESCRIPTION
**Pull request by RamanaReddy0M**

### CODE REVIEW

**Review by ShubhamRasal — CHANGES_REQUESTED** [review]
lgtm - suggesting small change

**Comment by tarunKoyalwar** [comment]
@RamanaReddy0M , providing a `example command/demo` helps in function testing and review
Ex:
```console
$ ./basic -h


Usage:
  ./basic [flags]

Flags:
INFO:
   -n, -name string     name of the user
   -e, -email string[]  email of the user

ADDITIONAL:
   -ph, -phone string       phone of the user
   -add, -address string[]  address of the user

UPDATE:
   -up, -update                 update nuclei to the latest released version
   -duc, -disable-update-check  disable automatic update check

```

Ref:
https://github.com/projectdiscovery/nuclei/pull/3331
https://github.com/projectdiscovery/tlsx/pull/177

File: update_group.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// NewUpdateGroup creates self-update options
+func (flagset *FlagSet) NewUpdateGroup(toolName string, uc UpdateCheckCallback, u UpdateCallback) {
+	ug := UpdateGroup{
+		Options: options{IsUpdate: false, DisableUpdate: false},
+		Actions: actions{UpdateCallback: u, UpdateCheckCallback: uc},
+	}
+	flagset.CreateGroup("update", "Update",
+		flagset.BoolVarP(&ug.Options.IsUpdate, "update", "up", false, fmt.Sprintf("update %v to the latest released version", toolName)),
+		flagset.BoolVarP(&ug.Options.DisableUpdate, "disable-update-check", "duc", false, "disable automatic update check"),
+	)
```
Comment (tarunKoyalwar) [inline update_group.go:33]: @RamanaReddy0M , suggesting multiple changes

- [ ] rename `NewUpdateGroup` -> `CreateUpdateGroup` which is more consistent
- [ ] `CreateUpdateGroup` should be uniform with other creategroup methods i.e should accept extra flags

```go
CrateUpdateGroup(toolName string, uc UpdateCheckCallback,u UpdateCallback,flags ...*goflags.FlagData)
```
- [ ] Rename Method parameters/options. its Final form should look like
```go
CreateUpdateGroup(toolName string,checkUpdate checkUpdateCallback, update UpdateCallback,flags ...*goflags.FlagData)
```

**Review by tarunKoyalwar — CHANGES_REQUESTED** [review]

**Comment by tarunKoyalwar** [comment]
closing this PR after discussion with @RamanaReddy0M  @Mzack9999 "Implementing CallbackVars seems better and simpler than New group type"

**Pull request closed without merging**